File: hello-kafka-api/java/src/main/java/hellokafka/Producer.java
package hellokafka;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;

@Service
@EnableScheduling
public class Producer {

    @Autowired
    public final KafkaTemplate<String, String> kafka = null;

    @Scheduled(fixedRate = 1000)
    public void send() {
        String message = "hello kafka  " + LocalDateTime.now().toString();
        System.out.println("SEND    MESSAGE : " + message );

        this.kafka.send("hello-topic", message);
    }
}

File: hello-kafka-api/kotlin/src/main/kotlin/hellokafka/Producer.kt
package hellokafka


import org.springframework.kafka.core.KafkaTemplate
import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Service
import java.time.LocalDateTime

@Service
@EnableScheduling
class Producer(val kafka: KafkaTemplate<String, String>) {

    @Scheduled(fixedRate = 1000)
    fun send() {
        val message: String = "hello kafka  " + LocalDateTime.now().toString()
        println("SEND    MESSAGE : $message")

        this.kafka.send("hello-topic", message)
    }
}
